metrics/src/counter.rs

```diff
@@ -230,36 +164,29 @@ impl Counter {
                 self.name,
                 counts + events,
                 times,
-                timing::timestamp(),
+                now,
                 events,
             );
         }
 
-        if times % metricsrate == 0 && times > 0 {
-            let lastlog = self.lastlog.load(Ordering::Relaxed);
-            let prev = self
-                .lastlog
-                .compare_and_swap(lastlog, counts, Ordering::Relaxed);
-            if prev == lastlog {
-                if let Some(ref mut point) = self.point {
-                    point
-                        .fields
-                        .entry("count".to_string())
-                        .and_modify(|v| {
-                            *v = influxdb::Value::Integer(counts as i64 - lastlog as i64)
-                        })
-                        .or_insert(influxdb::Value::Integer(0));
-                }
-                if let Some(ref mut point) = self.point {
-                    submit(point.to_owned(), level);
-                }
-            }
+        let lastlog = self.lastlog.load(Ordering::Relaxed);
+        let prev = self
+            .lastlog
+            .compare_and_swap(lastlog, counts, Ordering::Relaxed);
+        if prev == lastlog {
+            let bucket = now / metricsrate;
+            let counter = CounterPoint {
```

Comment:
I think we found allocating like this is kind of slow, hence we are using the cached point, is that possible here?

Reply:
I guess maybe not since we are sending to a channel.

Reply:
Oh interesting, are there differences in allocation performance between cloning an existing struct and allocating a new one?

Reply:
Anyways, we are allocating less with this change. `CounterPoint` is much lighter weight than `influxdb::Point`